### PR TITLE
Add serviceaccount to prod GCS buckets

### PR DESCRIPTION
**Pull request by saschagrunert**
The serviceaccount can be used for automation purposes, for example in GitHub actions when publishing releases.

Fixes https://github.com/kubernetes/k8s.io/issues/5957

### CODE REVIEW

File: infra/gcp/bash/ensure-prod-storage.sh [hunk 1]
```diff
@@ -250,13 +251,28 @@ function ensure_all_prod_projects() {
...
 function ensure_all_prod_buckets() {
+    local sa_name="k8s-infra-push-bot"
+    local sa_email="${sa_name}@${PROD_PROJECT}.iam.gserviceaccount.com"
+    local principal="serviceAccount:${sa_email}"
+
+    color 6 "Ensuring ${sa_email} exists and can write to prod buckets in project: ${PROD_PROJECT}"
+    ensure_service_account \
+      "${PROD_PROJECT}" \
+      "${sa_name}" \
+      "used by automation to push artifacts to prod buckets in ${PROD_PROJECT}"
```
Comment (saschagrunert) [inline infra/gcp/bash/ensure-prod-storage.sh:262]: @ameukam @xmudrii is there any automation required to get the service account private keys? I assume we manually create them and push them into the 1Password vault, right?

Comment (ameukam) [reply infra/gcp/bash/ensure-prod-storage.sh:262]: We prevent private keys creation. Workload Identity or Service Account impersonation are the only things allowed now.

Comment (saschagrunert) [reply infra/gcp/bash/ensure-prod-storage.sh:264]: Switched to workload identity as described in https://github.com/google-github-actions/auth

**Comment by saschagrunert** [comment]
@xmudrii @ameukam PTAL

**Comment by ameukam** [comment]
@BenTheElder @upodroid Can you PTAL ? 🙏🏾 
Limited bandwith on my size.

**Pull request closed without merging**